src/Sylius/Component/Core/Resolver/DefaultShippingMethodResolver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Resolver;
+
+use Sylius\Component\Shipping\Repository\ShippingMethodRepositoryInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class DefaultShippingMethodResolver implements DefaultShippingMethodResolverInterface
+{
+    /**
+     * @var ShippingMethodRepositoryInterface
+     */
+    private $shippingMethodRepository;
+
+    /**
+     * @param ShippingMethodRepositoryInterface $shippingMethodRepository
+     */
+    public function __construct(ShippingMethodRepositoryInterface $shippingMethodRepository)
+    {
+        $this->shippingMethodRepository = $shippingMethodRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultShippingMethod()
+    {
+        $shippingMethods = $this->shippingMethodRepository->findAll();
```

Comment:
How about using `ShopperContext` to get current channel and use `findEnabledForChannel` repository method?